planner/core/point_get_plan_test.go

```diff
@@ -600,3 +600,14 @@ func (s *testPointGetSuite) TestCBOShouldNotUsePointGet(c *C) {
 		res.Check(testkit.Rows(output[i].Res...))
 	}
 }
+
+func (s *testPointGetSuite) TestIssue18042(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table t")
+	tk.MustExec("create table t(a int, b int, c int, primary key(a));")
+	tk.MustExec("insert into t values (1, 1, 1), (2, 2, 2), (3, 3, 3), (4, 4, 4)")
+	tk.MustExec("SELECT /*+ MAX_EXECUTION_TIME(100)*/ * FROM t where a = 1;")
```

Comment:
please add more test for other hints, like `use_index()`

Reply:
Index hints like `use_index` seem trivial to fast plans like point-get, while runtime hints like exec-time or mem-quota are more important.
I added mem-quota into this test, PTAL.

Reply:
Maybe we need to determine the behavior of `use_index()` with `point get`. For example, when we specify an index that cannot use `point get`, what will the optimizer do?

Reply:
Yes~ `use_index()` can be applied to `point get`, but I think it's trivial.
For example, for the query `select a from t where a=1`, using `index(a)` or `index(a, b)` as its access path in a `point get` has the same efficiency.
And `point get` doesn't support `use_index()` now, so we have to introduce more lines of code into `point get` if we want to support and test it.
On second thought, I think it's better to doesn't change it and keep the status quo.